pulser/simulation/simresults.py

```diff
@@ -15,15 +15,20 @@
 import qutip
 import numpy as np
 
+from collections import Counter
+from abc import ABC, abstractmethod
 
-class SimulationResults:
-    """Results of a simulation run of a pulse sequence.
+
+class SimulationResults(ABC):
+    """Results of a simulation run of a pulse sequence. Parent class for
+    NoisyResults and CleanResults.
```

Comment:
We follow [Google's style guide](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings), which states this must be a one-line summary. I would suggest moving the parent class information to the block below.